eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/ReplicatedCassandra.scala

```diff
@@ -455,13 +481,15 @@ private[journal] object ReplicatedCassandra {
                                   .to[F] {
                                     to
                                       .next[Option]
-                                      .getOrElse { journalHead.seqNr }
+                                      .getOrElse { headSeqNr }
                                       .toSegmentNr(segmentSize)
                                   }
-                                _ <- segmentNrs.parFoldMapA { segmentNr =>
-                                  statements
-                                    .deleteRecords(key, segmentNr)
-                                    .uncancelable
+                                _ <- Sync[F].whenA(from < to) {
+                                  segmentNrs.parFoldMapA { segmentNr =>
+                                    statements
+                                      .deleteRecords(key, segmentNr)
+                                      .uncancelable
+                                  }
```

Comment:
extra optimization to not issue delete statements when `from == to` - original unit-test had a unit-test, which kind of expected that no `delete` statements against `journal` table are done, while they were actually done, though not recorded in `StateT`

Reply:
Please elaborate why events should not be deleted if `from == to`. I see it wrong because:

- purge is rare operation where optimizations not so important
- data inconsistency can lead to orphan events thus deleting them explicitly is fine

Reply:
BTW, purge is not rare - we run it every night to remove all expired journals - `PurgeExpired` finds all records on `metajournal` which have expired and then generates `Purge` action for each of them - every night there is a flood of `Purge` actions.

Reply:
how come this change is "fixing test"? I see it as fixing code because there's some test ... Please revert it